esmvaltool/diag_scripts/ocean/diagnostic_maps.py

```diff
@@ -105,11 +150,247 @@ def make_map_plots(
         plt.close()
 
 
+def make_map_contour(
+        cfg,
+        metadata,
+        filename,
+):
+    """
+    Make a simple contour map plot for an individual model.
+
+    Parameters
+    ----------
+    cfg: dict
+        the opened global config dictionairy, passed by ESMValTool.
+    metadata: dict
+        the metadata dictionairy
+    filename: str
+        the preprocessed model file.
+
+    """
+    # Load cube and set up units
+    cube = iris.load_cube(filename)
+    cube = diagtools.bgc_units(cube, metadata['short_name'])
+
+    # Is this data is a multi-model dataset?
+    multi_model = metadata['dataset'].find('MultiModel') > -1
+
+    # Make a dict of cubes for each layer.
+    cubes = diagtools.make_cube_layer_dict(cube)
+
+    # Load image format extention and threshold.thresholds.
+    image_extention = diagtools.get_image_format(cfg)
+
+    # Load threshold/thresholds.
+    plot_details = {}
+    colours = []
+    thresholds = diagtools.load_thresholds(cfg, metadata)
+
+    for itr, thres in enumerate(thresholds):
+        if len(thresholds) > 1:
+            colour = plt.cm.jet(float(itr) / float(len(thresholds) - 1.))
+        else:
+            colour = plt.cm.jet(0)
+        label = str(thres) + ' ' + str(cube.units)
+        colours.append(colour)
+        plot_details[thres] = {'c': colour,
+                               'lw': 1,
+                               'ls': '-',
+                               'label': label}
+
+    linewidths = [1 for thres in thresholds]
+    linestyles = ['-' for thres in thresholds]
+    # Making plots for each layer
+    for layer_index, (layer, cube_layer) in enumerate(cubes.items()):
+        layer = str(layer)
+        # cube_layer = regrid_irregulars(cube_layer)
+        qplt.contour(cube_layer,
+                     thresholds,
+                     colors=colours,
+                     linewidths=linewidths,
+                     linestyles=linestyles,
+                     rasterized=True)
+
+        try:
+            plt.gca().coastlines()
+        except AttributeError:
+            logger.warning('Not able to add coastlines')
+        try:
+            plt.gca().add_feature(cartopy.feature.LAND,
+                                  zorder=10,
+                                  facecolor=[0.8, 0.8, 0.8])
+        except AttributeError:
+            logger.warning('Not able to add coastlines')
+        # Add legend
+        diagtools.add_legend_outside_right(plot_details,
+                                           plt.gca(),
+                                           column_width=0.02,
+                                           loc='below')
+
+        # Add title to plot
+        title = ' '.join([metadata['dataset'], metadata['long_name']])
+        if layer:
+            title = ' '.join(
+                [title, '(', layer,
```

Comment:
it be nicer if you used string formatting `'blah {} boo {}'.format('hum', 'bug')`

Reply:
Done